### PR TITLE
Swapped from TravisCI to GitHub Actions

### PR DESCRIPTION
**Pull request by djspiewak**
This mirrors the way the TravisCI build worked. We need the following three secrets configured in the project:

- `PRIVATE_KEY_PASSWORD`
- `PRIVATE_KEY_IV`
- `SONATYPE_PASS`

Also I need to push an update which makes that last one work. But the others are good to go.

### CODE REVIEW

**Comment by djspiewak** [comment]
@jodersky I think this is good to go! Just needs the secrets in the OP. Also you need to enable GitHub Actions for this repository. Beyond that, everything should be all good.

**Comment by jodersky** [comment]
Looks good! One thing to note however, I have lost the original passphrase used to encrypt the GPG signing key. I probably do have a backup on my other machine, but I think it's a good idea to revise this setup anyway.

Sonatype still requires a key, but since we're no longer on travis, maybe we can simplify the setup: instead of encrypting the key in a file and decrypting it as part of CI, could the key be stored directly in an environment variable? Concretely, instead of storing an encrypted file along with the two secrets `PRIVATE_KEY_PASSWORD` and `PRIVATE_KEY_IV`, do you think it would be preferable to store the key directly in a secret `GPG_SECRET_KEY`?

**Comment by jodersky** [comment]
If the above works for you, I have created two secrets in this repo:
1. `SONATYPE_PASS` the API key for publishing to sonatype
2. `GPG_SECRET_KEY` the ascii-armored secret key of `0xA7A5BFE670846603`, which I just generated. In case anyone is interested, the public key is below and has also been uploaded to the ubuntu key servers.
```
-----BEGIN PGP PUBLIC KEY BLOCK-----

mQINBGEvp8oBEADCenKJ9eewd/icBrRYSlfq3yVTPGJ+WsBQLa7QbMUxvioKKyFH
z9XWrogD2zDDee3NaY3y/Bsq+Q6ECC85hz3Ii9CeYOAiQGZalI5kaKyDCCG8xfbJ
+9HdUdY7dEx/6vDByL3gWYqF8b3EkH2b5rEpHBm7lKuHQsXll9RyAAHjkgLxXy2R
B9xQy2KxJtJSllqTSsf84FSTgHB5ehdhT/mv93vmT3BoNP6iVraCmR25XxiCiMAM
KLxTRcaw4+lG3+ODKw/rPkvuH6KF6W+87ukIg81jtMEaJwLQX2Ctu0m1nqIbvSgB
ADzLr3jujrHUTDC9TxGNRf0HDpX7ILZHlmWOrgu4m5EdyoZyBtw1DRAWRURIIOHf
pyjGMeVPRZuyoai2GlJoLE99t0s7Bxa5UAyEml2wR82wD/mcXvMEMqLsN51Yd57Y
0z3qa/TB/Ce0kx9/mr54HLOI+KPCLpGYKee/kuPhgNgX2O/QPh/fUNE8s6xAOQ1w
gdEZfF5kwp1IcwfG0/WcVpTIxxA/fWl0pJ6euJD2f+jxlD7GEalbChP2jOESloOj
zSuW05L2Nd7I/xAmkiBA2f6Z5aJkTFhdNNKvYcu8I9mT/bTx+65/zQK5tiqSP839
ZKQJKYmDcd0LrIKd1Psx1+lLGnQGR0xXHtP5qFKuvBuaCs+n6STUOB96OQARAQAB
tBtHaXRIdWIgQWN0aW9ucyBmb3Igam9kZXJza3mJAk4EEwEKADgWIQQI4z2qyb9J
4b9UWGynpb/mcIRmAwUCYS+nygIbAwULCQgHAwUVCgkICwUWAgMBAAIeAQIXgAAK
CRCnpb/mcIRmA5JAD/9Wle+xyRAbpsBKG7gftcb+UcVn1BTpTnVtz2dxQv3q9EIV
8c1yWrbyCrGpQdqeofDUtSSI2nMfmo4t6Tx5vs/5flxexVMnPz58lCC8968zqnw1
bZqYfdYa7atfADJdNgKRYvhndrBcEpbmLrF4YncUtUWzjJ89lfv5AjL+gsRBX/ga
FF5ePoaXpYSYmsMbIOMvxhCHRd9e9n2FKlH3uOpNdHtiz8Sq/fjQN53huXoriGjB
rveS2/LnAuP+PWoppWY9eNz3eNfff0rrHAuj3KBOteZp323gQRm0e7h9uLc/JqHV
Eti2RQavCCRweDVNT8bJwCh1RKSr1NQ0e0B5oBnA+aP548sh5mK96w8AHjKwhn6h
jlm+b2ellMymSNYmsZ72vQvHAE6phiXQ51VhXETaNjE2EZcVJp3U0jQjy++s45hD
JEzIdD208RwaBmrFEffdU8DEFgpX2jD5g3FOfdNAd9gnzNCxVpnrLk0O0V340rAW
nt9p8FoifOSHs5LfYIl8dxp0k7lQSmOkRuqhFMMp7f2k1bJTdRkHJLZ62/2mldEG
wb7B9iPVyqagNr1287Uov9ZaSpD7AP5E4YfYqFBK9SCCOmqGFAwvES3S4/N1yjxt
NdEOIgHyp1sr5a+9Gih9HOs5U6suHzrnhWNDPN8ufidRKcSh1+7mvsgsoEQh0A==
=ekPF
-----END PGP PUBLIC KEY BLOCK-----
```

File: publish.sbt [hunk 1]
```diff
@@ -1,25 +1,50 @@
...
 )
+
+ThisBuild / githubWorkflowTargetTags ++= Seq("v*")
+
+ThisBuild / githubWorkflowPublishTargetBranches :=
+  Seq(RefPredicate.StartsWith(Ref.Tag("v")))
+
+ThisBuild / githubWorkflowPublishPreamble ++= Seq(
+  WorkflowStep.Run(
+    List("echo $GPG_SECRET_KEY | gpg --import"),
```
Comment (jodersky) [inline publish.sbt]: does $GPG_SECRET_KEY need quotes here?

Comment (djspiewak) [reply publish.sbt]: Ah, yes it does. Good catch!

**Review by jodersky — APPROVED** [review]
The GPG_KEY needs to be quoted, I think. But apart from that, LGTM! Thank you for this

**Pull request merged**